test/format/ops.c

```diff
@@ -481,6 +452,51 @@ snap_check(WT_CURSOR *cursor,
 	return (0);
 }
 
+/*
+ * begin_transaction --
+ *	Choose an isolation configuration and begin a transaction.
+ */
+static void
+begin_transaction(TINFO *tinfo, WT_SESSION *session, u_int *iso_configp)
+{
+	u_int v;
+	const char *config;
+	char config_buf[64];
+
+	if ((v = g.c_isolation_flag) == ISOLATION_RANDOM)
+		v = mmrand(&tinfo->rnd, 2, 4);
+	switch (v) {
+	case ISOLATION_READ_UNCOMMITTED:
+		config = "isolation=read-uncommitted";
+		break;
+	case ISOLATION_READ_COMMITTED:
+		config = "isolation=read-committed";
+		break;
+	case ISOLATION_SNAPSHOT:
+	default:
+		v = ISOLATION_SNAPSHOT;
+		config = "isolation=snapshot";
+		if (g.c_txn_timestamps) {
+			/*
+			 * Update the thread's read timestamp with the current
+			 * value to prevent the oldest timestamp moving past our
+			 * allocated timestamp before the commit completes.
+			 */
+			tinfo->read_timestamp = g.timestamp;
+			tinfo->read_timestamp =
+			    __wt_atomic_addv64(&g.timestamp, 1);
+			testutil_check(__wt_snprintf(
+			    config_buf, sizeof(config_buf),
+			    "read_timestamp=%" PRIx64, tinfo->read_timestamp));
+			config = config_buf;
+		}
+		break;
+	}
+	*iso_configp = v;
+
+	testutil_check(session->begin_transaction(session, config));
```

Comment:
It would be cleaner to clear `tinfo->read_timestamp` here, rather than in `ops`.  There's no reason to prevent the application's timestamp moving forward because WT already copes with the `oldest_timestamp` moving past a running query.

In other words `begin_transaction` consumes the `read_timestamp`, just like `commit_transaction` consumes the `commit_timestamp`.

Reply:
Both comments fixed in 4b5cf52.